browser/src/control/Control.UIManager.js

```diff
@@ -656,10 +657,21 @@ L.Control.UIManager = L.Control.extend({
 	},
 
 	showButton: function(buttonId, show) {
-		if (!this.notebookbar)
+		if (!this.notebookbar) {
 			this.showButtonInClassicToolbar(buttonId, show);
-		else
-			this.notebookbar.showShortcutsButton(buttonId, show);
+		} else {
+			if (show) {
+				delete this.hiddenButtons[buttonId];
+			} else {
+				this.hiddenButtons[buttonId] = true;
+			}
+			this.notebookbar.reloadShortcutsBar();
+			this.notebookbar.showButton(buttonId, show);
```

Comment:
it that order correct? : `first RELOAD then SHOW`

Reply:
actually the order doesn't matter. `reloadShortcutsBar` replace `showShortcutsButton` since we have `hiddenButtons`, while `showButton` is about notebook bar buttons. Maybe I should just rename it to `showNotebookButton`.

Also it should be noted that the shortcut buttons have different ID than the notebook buttons counterparts.

Reply:
Just to summarize:
- set the `hiddensButtons` to the right values
- call `reloadShortcutsBar` which will handle the shortcut bar.
- call `showButton` (that I'll rename to be less confusing) to show / hide the notebookbar buttons.

Reply:
I see! it's good then